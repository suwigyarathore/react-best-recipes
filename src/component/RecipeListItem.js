import React from 'react';
import PropTypes from 'prop-types';

function RecipeListItem({ recipe, favorited, onClick, onFavorited }) {
  return (
    <li
      className="py2 border-bottom border-bottom-dashed pointer"
      key={recipe.id}
      onClick={() => onClick(recipe.id)}
    >
      <span
        role="presentation"
        aria-label="favorite"
        className="mx1"
        onClick={e => {
          e.stopPropagation();
          onFavorited(recipe.id);
        }}
      >
        {favorited ? '✅' : '⬜️'}
      </span>
      <span>{recipe.name}</span>
      <span>{recipe.category}</span>
    </li>
  );
}

RecipeListItem.propTypes = {
  recipe: PropTypes.object,
  favorited: PropTypes.bool,
  onClick: PropTypes.func,
  onFavorited: PropTypes.func,
};

RecipeListItem.defaultProps = {
  recipe: {},
};

export default RecipeListItem;
