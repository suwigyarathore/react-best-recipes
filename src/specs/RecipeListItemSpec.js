import React from 'react';
import renderer from 'react-test-renderer';
import RecipeListItem from '../component/RecipeListItem';

const testRecipe = {
  id: 1,
  name: 'Test Recipe',
  category: 'Test Category',
};

describe('<RecipeListItem />', () => {
  test('Should not break if no recipe passed ', () => {
    const component = renderer.create(<RecipeListItem />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('Should correctly render recipe', () => {
    const component = renderer.create(<RecipeListItem recipe={testRecipe} />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('Should render favorite state in list', () => {
    const component = renderer.create(
      <RecipeListItem recipe={testRecipe} favorited />,
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
