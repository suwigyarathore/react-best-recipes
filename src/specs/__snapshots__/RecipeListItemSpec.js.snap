// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RecipeListItem /> Should correctly render recipe 1`] = `
<li
  className="py2 border-bottom border-bottom-dashed pointer"
  onClick={[Function]}
>
  <span
    aria-label="favorite"
    className="mx1"
    onClick={[Function]}
    role="presentation"
  >
    ⬜️
  </span>
  <span>
    Test Recipe
  </span>
  <span>
    Test Category
  </span>
</li>
`;

exports[`<RecipeListItem /> Should not break if no recipe passed  1`] = `
<li
  className="py2 border-bottom border-bottom-dashed pointer"
  onClick={[Function]}
>
  <span
    aria-label="favorite"
    className="mx1"
    onClick={[Function]}
    role="presentation"
  >
    ⬜️
  </span>
  <span />
  <span />
</li>
`;

exports[`<RecipeListItem /> Should render favorite state in list 1`] = `
<li
  className="py2 border-bottom border-bottom-dashed pointer"
  onClick={[Function]}
>
  <span
    aria-label="favorite"
    className="mx1"
    onClick={[Function]}
    role="presentation"
  >
    ✅
  </span>
  <span>
    Test Recipe
  </span>
  <span>
    Test Category
  </span>
</li>
`;
